packages/protocol/contracts/identity/Random.sol

```diff
@@ -93,20 +95,33 @@ contract Random is IRandom, Ownable, Initializable {
    */
   function addRandomness(uint256 blockNumber, bytes32 randomness) internal {
     history[blockNumber] = randomness;
-    if (historySize == 0) {
-      historyFirst = block.number;
-      historySize = 1;
-    } else if (historySize > randomnessBlockRetentionWindow) {
-      delete history[historyFirst];
-      delete history[historyFirst + 1];
-      historyFirst += 2;
-      historySize--;
-    } else if (historySize == randomnessBlockRetentionWindow) {
-      delete history[historyFirst];
-      historyFirst++;
+    if ((blockNumber + 1) % getEpochSize() == 0) {
```

Comment:
Why `+1`?

Reply:
Confusion over epoch indexing - to be clear for others: epochs are arranged by blocks as:
block 0 : epoch 0
block 1 - 100 (inclusive): epoch 1
block 101 - 200 (inclusive): epoch 2
Thus the last block of any epoch is a multiple of epoch_size, so an epoch's block number is sort-of 1-indexed. The typescript testing utils seem to be built aware of the 1-indexing, but not of the 0'th epoch, thus it treats block 1 as in epoch 0, and block 0 would break `currentEpochNumber`.
Luckily our tests only seem to use `currentEpochNumber` as a relative measurement, but I'll create an issue to fix this.

Reply:
I think @jfoutts-celo has pointed this out as well

Reply:
Created Issue #2718 Epoch number definition mismatch, to track this.